src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py

```diff
@@ -347,6 +349,9 @@ def ensure_container_insights_for_monitoring(
     if not addon.enabled:
         return None
 
+    if (not is_private_cluster or not aad_route) and azure_monitor_private_link_scope_resource_id is not None:
+        raise AzCLIError("--azure-monitor-private-link-scope-resource-id can only be used with private cluster in MSI mode.")
```

Comment:
Could we use the specific error type `InvalidArgumentValueError` instead of `AzCLIError`?

Reply:
> Could we use the specific error type `ArgumentUsageError` instead of `AzCLIError`?

Fixed.